src/benchmarks/micro/libraries/System.Text.RegularExpressions/Perf.Regex.Industry.cs

```diff
@@ -0,0 +1,302 @@
+            @"\s[a-zA-Z]{0,12}ing\s"
+        )]
+        public string Pattern { get; set; }
+
+        [Params(
+            RegexOptions.None,
+            RegexOptions.Compiled
+#if NET7_0_OR_GREATER
+            , RegexOptions.NonBacktracking
+#endif
+            )]
+        public RegexOptions Options { get; set; }
+
+        private string _sherlock;
+        private Regex _regex;
+
+        [GlobalSetup(Target = nameof(Count))]
+        public void Setup()
+        {
+            _regex = new Regex(Pattern, Options);
+            _sherlock = Perf_Regex_Industry.ReadInputFile("sherlock.txt.gz");
+        }
+
+        [Benchmark]
+        public int Count() => Perf_Regex_Industry.Count(_regex, _sherlock);
+    }
+
+    /// <summary>Performance tests adapted from https://rust-leipzig.github.io/regex/2017/03/28/comparison-of-regex-engines/</summary>
+    [BenchmarkCategory(Categories.Libraries)]
+    [SkipTooManyTestCasesValidator]
+    public class Perf_Regex_Industry_Leipzig
+    {
+        [Params(
+            "Twain",
+            "(?i)Twain",
+            "[a-z]shing",
+            "Huck[a-zA-Z]+|Saw[a-zA-Z]+",
+            //"\\b\\w+nn\\b", // duplicates (almost) Perf_Regex_Industry_RustLang_Sherlock test
+            //"[a-q][^u-z]{13}x", // duplicates Perf_Regex_Industry_RustLang_Sherlock test
+            "Tom|Sawyer|Huckleberry|Finn",
+            "(?i)Tom|Sawyer|Huckleberry|Finn",
+            ".{0,2}(Tom|Sawyer|Huckleberry|Finn)",
+            ".{2,4}(Tom|Sawyer|Huckleberry|Finn)",
+            "Tom.{10,25}river|river.{10,25}Tom",
+            //"[a-zA-Z]+ing", // duplicates Perf_Regex_Industry_RustLang_Sherlock test
+            //"\\s[a-zA-Z]{0,12}ing\\s", // duplicates Perf_Regex_Industry_RustLang_Sherlock test
+            "([A-Za-z]awyer|[A-Za-z]inn)\\s",
+            //"[\"'][^\"']{0,30}[?!\\.][\"']", // Breaks benchmarkdotnet 13.1
+            "\u221E|\u2713",
+            "\\p{Sm}")]
+        public string Pattern { get; set; }
+
+        [Params(
+            RegexOptions.None,
+            RegexOptions.Compiled
+#if NET7_0_OR_GREATER
+            , RegexOptions.NonBacktracking
+#endif
+            )]
+        public RegexOptions Options { get; set; }
+
+        private string _3200;
+        private Regex _regex;
+
+        [GlobalSetup(Target = nameof(Count))]
+        public void Setup()
+        {
+            _regex = new Regex(Pattern, Options);
+            _3200 = Perf_Regex_Industry.ReadInputFile("3200.txt.gz");
+        }
+
+        [Benchmark]
+        public int Count() => Perf_Regex_Industry.Count(_regex, _3200);
+    }
+
+    /// <summary>Performance tests adapted from https://www.boost.org/doc/libs/1_41_0/libs/regex/doc/gcc-performance.html</summary>
+    [BenchmarkCategory(Categories.Libraries)]
+    [SkipTooManyTestCasesValidator]
+    public class Perf_Regex_Industry_BoostDocs_Simple
+    {
+        [Params(0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13)]
+        public int Id { get; set; }
```

Comment:
I like the idea of using Ids to avoid the need of having very long arguments printed in the output and included in the Benchmark ID 👍